### PR TITLE
Made log_softmax function support  complex dtype

### PR DESCRIPTION
**Pull request by mohame54**

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/backends/torch/activations.py [hunk 1]
```diff
@@ -107,23 +107,26 @@ def softsign(x: torch.Tensor, /, out: Optional[torch.Tensor] = None) -> torch.Te
...
 ):
     if axis is None:
         axis = -1
+    if torch.is_complex(x):
+        x_max = torch_backend.max(x, axis=axis, keepdims=True)
+        sub_temp = torch.sub(x, x_max)
+        ret = torch.sum(sub_temp.exp(), dim=axis, keepdim=True)
+        ret = torch.log(ret)
+        ret = torch.sub(sub_temp, ret)
+        return ret
```
Comment (jshepherd01) [inline ivy/functional/backends/torch/activations.py:129]: Since softmax is already implemented why not use `torch.log(torch_backend.softmax(x))`?

**Review by jshepherd01 — COMMENTED** [review]
Looks good, just a question about the implementation

File: ivy/functional/backends/torch/activations.py (hunk 1 above)
Comment (mohame54) [reply ivy/functional/backends/torch/activations.py:129]: at first I thought about this approach but when I considered torch documentation about **log_softmax**  that doing **log** and **softmax* ops is computationally expensive also not numerically stable I though of a second implementation which is faster, simpler and also mathematically equivalent. here
log_softmax(x) =log(exp(x)/ sum(exp(x)) => x - log(sum(exp(x))

Comment (mohame54) [reply ivy/functional/backends/torch/activations.py:129]: image
Here is the torch doc

Comment (jshepherd01) [reply ivy/functional/backends/torch/activations.py:129]: Oh cool, that looks good then

**Review by jshepherd01 — APPROVED** [review]
Okay, this looks good to me. There are a couple of test failures but I think that's due to the testing pipeline's bfloat16 issues and tensorflow's general poor performance with complex numbers, could you take care of the merge conflict and then I'd be happy to merge

**Comment by mohame54** [comment]
okay no problem
